### PR TITLE
image-partition: Enable filesystem features on fs creation

### PR DESCRIPTION
**Pull request by a-wai**
We might want to enable specific features during a filesystem creation,
as it can be difficult and/or cause issues to enable them afterwards.

This commit introduces a new 'features' field to partition definitions,
and makes sure the corresponding features are enabled when creating the
filesystem.

### CODE REVIEW

File: actions/image_partition_action.go [hunk 1]
```diff
@@ -179,6 +184,21 @@ func (p *Partition) UnmarshalYAML(unmarshal func(interface{}) error) error {
...
+func appendFeatures(cmdline []string, features []string) []string {
+	if len(features) > 0 {
+		var tmp strings.Builder
+		for _, o := range features {
+			fmt.Fprintf(&tmp, "%s,", o)
+		}
+		options := tmp.String()
+		// Strip final ","
+		options = options[:tmp.Len()-1]
+		cmdline = append(cmdline, "-O", options)
```
Comment (sjoerdsimons) [inline actions/image_partition_action.go]: You can use strings.Join to concatinate the list with ',' as seperator; that probalby cuts away half of the code :)

Comment (sjoerdsimons) [reply actions/image_partition_action.go]: Also can Features not be nil ? I'd expect taht to be the default

Comment (a-wai) [reply actions/image_partition_action.go]: > You can use strings.Join to concatinate the list with ',' as seperator; that probalby cuts away half of the code :)

Indeed, thanks for the hint!

> Also can Features not be nil ? I'd expect taht to be the default

I'm not sure what you mean here: Features is a member of the `Partition` struct, can it be nil in a valid `Partition` instance ?

**Comment by gportay** [comment]
What is the status of that PR?

**Pull request merged**